src/connector/codec/src/decoder/json/mod.rs

```diff
@@ -12,11 +12,170 @@
+                                reqwest::get(ref_url_no_fragment)
+                                    .await
+                                    .into_request(&ref_no_fragment)?
+                                    .json()
+                                    .await
+                                    .into_request(&ref_no_fragment)?
+                            } else if url_schema == "file" {
+                                let file_path =
+                                    ref_url_no_fragment.to_file_path().map_err(|_| {
+                                        anyhow::anyhow!(
+                                            "could not convert url {} to file path",
+                                            &ref_url_no_fragment
+                                        )
+                                    })?;
+                                let file = fs::File::open(file_path)
+                                    .into_schema_from_file(&ref_no_fragment)?;
+                                serde_json::from_reader(file)
+                                    .into_schema_not_json_serde(ref_no_fragment.clone())?
+                            } else {
+                                return Err(Error::UnsupportedUrl {
+                                    url: ref_no_fragment,
+                                });
+                            }
+                        }
+                    };
+
+                    if !self.schema_cache.contains_key(&ref_no_fragment) {
+                        self.schema_cache
+                            .insert(ref_no_fragment.clone(), schema.clone());
+                    }
+
+                    let ref_url_string = ref_url.to_string();
+                    if let Some(ref_fragment) = ref_url.fragment() {
+                        schema = schema
+                            .pointer(ref_fragment)
+                            .ok_or(Error::JsonRefPointerNotFound {
+                                ref_string: ref_string.to_owned(),
+                                ref_fragment: ref_fragment.to_owned(),
+                            })?
+                            .clone();
+                    }
+                    // Do not deref a url twice to prevent infinite loops
+                    if used_refs.contains(&ref_url_string) {
+                        return Ok(());
+                    }
+                    let mut new_used_refs = used_refs.clone();
+                    new_used_refs.push(ref_url_string);
+
+                    Box::pin(self.deref(&mut schema, ref_no_fragment, &new_used_refs)).await?;
+                    let old_value = mem::replace(value, schema);
+
+                    if let Some(reference_key) = &self.reference_key {
+                        if let Some(new_obj) = value.as_object_mut() {
+                            new_obj.insert(reference_key.clone(), old_value);
+                        }
+                    }
```

Comment:
Any document on this behavior? It seems `self.reference_key` is constantly set to `Some("__reference__".to_owned())`

Reply:
Yes it's unccessary.  It's just when a object contains more than `$ref`. The items other than `$ref` will be ignored and put it in a __reference__ file. It's from the python and rust library `jsonref`

Reply:
Seem the dereferencing implementation is from the `jsonref` library (including the repeated unnecessary `parse` + `to_string`), so we need to include its MIT license.

Let's also remove the unused `__reference__` key to avoid confusing people reading this part. As a standalone library it provides flexibility to the caller, but it is just dead code here.

Reply:
> so we need to include its MIT license.

Good catch. Can't believe I missed this...

> remove the unused __reference__ key

done